gdal/ogr/ogrsf_frmts/geojson/ogrgeojsonwriter.cpp

```diff
@@ -79,6 +79,7 @@ void OGRGeoJSONWriteOptions::SetIDOptions(CSLConstList papszOptions)
             eForcedIDFieldType = OFTInteger64;
         }
     }
+    bGenerateID = CSLFetchBoolean(papszOptions, "ID_GENERATE", false);
```

Comment:
With MSVC 2015: c:\projects\gdal\gdal\ogr\ogrsf_frmts\geojson\ogrgeojsonwriter.cpp(82): warning C4800: 'int': forcing value to bool 'true' or 'false' (performance warning)
Try this:
bGenerateID = CPL_TO_BOOL(CSLFetchBoolean(papszOptions, "ID_GENERATE", false));